conan/tools/gnu/autotoolsdeps.py

```diff
@@ -4,66 +4,79 @@
 from conans.model.new_build_info import NewCppInfo
 
 
+def _get_cpp_info(deps):
+    ret = NewCppInfo()
+    for dep in deps:
+        dep_cppinfo = dep.cpp_info.aggregated_components()
+        # In case we have components, aggregate them, we do not support isolated
+        # "targets" with autotools
+        ret.merge(dep_cppinfo)
+    return ret
+
+
+def _rpaths_flags(deps):
+    flags = []
+    for dep in deps:
+        flags.extend(["-Wl,-rpath -Wl,{}".format(libdir) for libdir in dep.cpp_info.libdirs
+                      if dep.options.get_safe("shared", False)])
+    return flags
+
+
+def ordered_deps(conanfile):
+    deps = conanfile.dependencies.host.topological_sort
+    return[dep for dep in reversed(deps.values())]
+
+
+def get_gnu_deps_flags(conanfile):
```

Comment:
a free function probably doen't belong to `AutoToolsDeps` generator, it should be somewhere else in `conan.tools`?

Reply:
The logic was already here, so I've kept it here as well. I don't think that it has to be in Conan tools but I'm not saying that this is the better place or even if we want it. By the record, the final import is `from conan.tools.gnu import get_gnu_deps_flags` so I guess it makes sense... Let's discuss! 😄